src/middlewares/auth/authCustom.js

```diff
@@ -0,0 +1,63 @@
+const { User, UserStatuses, UserRoles } = require('../../models');
+const bcrypt = require('bcryptjs');
+const passport = require('passport');
+const CustomStrategy = require('passport-custom').Strategy;
+const GoogleStrategy = require('passport-google-oauth2').Strategy;
+require('dotenv').config();
+
+const verify = async (req, done) => {
+  const checkUser = await User.findOne({ email: req.body.email }).exec();
+
+  if (!checkUser) {
+    return done(null, false);
+  }
+
+  const status = await UserStatuses.findOne({ name: 'Active' }).exec();
+
+  if (!(checkUser.status.toString() === status._id.toString())) {
+    return done(null, false);
+  }
+
+  const result = await bcrypt.compare(req.body.password, checkUser.password);
+
+  if (!result) {
+    done(null, false);
+  }
+
+  const user = JSON.parse(JSON.stringify(checkUser));
+  delete user.password;
+  delete user.address;
+  delete user.phoneNumber;
+  delete user.orders;
+
+  done(null, user);
+};
+
+passport.use('custom', new CustomStrategy(verify));
+
```

Comment:
Name this file more appropriate... authStrategies for example

Reply:
All these connects passport.use etc must be in main index file. Leave here only strategies

Reply:
Fixed.